### PR TITLE
Added an abyss component to flowcraft, and added a new disableRR parameter to spades

### PR DESCRIPTION
**Pull request by emreerhan**

### CODE REVIEW

File: flowcraft/generator/components/assembly.py [hunk 1]
```diff
@@ -177,3 +182,39 @@ def __init__(self, **kwargs):
...
+    """
+
+    def __init__(self, **kwargs):
+
+        super().__init__(**kwargs)
+
+        self.input_type = "fastq"
+        self.output_type = "fasta"
+
+        # self.dependencies = ["integrity_coverage"]
```
Comment (ODiogoSilva) [inline flowcraft/generator/components/assembly.py]: It's OK to remove this line altogether

File: flowcraft/generator/components/assembly.py [hunk 2]
```diff
@@ -177,3 +177,39 @@ def __init__(self, **kwargs):
...
```
Comment (tiagofilipe12) [inline flowcraft/generator/components/assembly.py]: Do you need this line? If you don't need it, remove it.

File: flowcraft/generator/engine.py [hunk 3]
```diff
@@ -101,7 +101,8 @@
         "split_assembly": meta.SplitAssembly,
         "trimmomatic": readsqc.Trimmomatic,
         "true_coverage": readsqc.TrueCoverage,
-        "viral_assembly": assembly.ViralAssembly
+        "viral_assembly": assembly.ViralAssembly,
+        "abyss": assembly.Abyss
```
Comment (tiagofilipe12) [inline flowcraft/generator/engine.py]: Although this will probably be refactored according to #145 , can you keep the alphabetical order for now?

**Comment by emreerhan** [comment]
I accidentally pushed my changes to `spades` to this fork. I'll edit the title of the PR to reflect this. Is there anything else I should do too?

**Comment by ODiogoSilva** [comment]
Yes, there is one more thing. Could add these changes to the `changelog.md`? The new components should be under a header of `### New components` and the changes to existing components with `### Components changes`

File: flowcraft/generator/components/assembly.py (hunk 2 above)
Comment (emreerhan) [reply flowcraft/generator/components/assembly.py]: I'll remove it!

File: flowcraft/generator/engine.py (hunk 3 above)
Comment (emreerhan) [reply flowcraft/generator/engine.py]: Done

**Comment by ODiogoSilva** [comment]
Sooooorry @emreerhan :sweat_smile: Could you put the change log entries in the `Changes in upcoming release (dev branch)`? They are currently under the 1.3.1 release, which already occurred. You'll need to add the headers as well.

**Comment by emreerhan** [comment]
Oops!

**Comment by ODiogoSilva** [comment]
Welcome to the team  @emreerhan :muscle: The docker image for abyss will also be up in dockerhub shortly, after the PR is merged.

**Comment by tiagofilipe12** [comment]
Good job and welcome to the team @emreerhan . Very nice PR. Hope you have fun with FlowCraft!

**Pull request merged**